src/main.cpp

```diff
@@ -6005,6 +6005,9 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
         {
             boost::this_thread::interruption_point();
 
+            if (!(addr.nServices & NODE_NETWORK))
+                continue;
+
```

Comment:
For this commit that you removed I'm wondering if maybe the better approach would be to not share addresses for anything that does NOT have a service at all.  That way we would still share XTHIN nodes but keep out all the spam nodes that don't have services.  Still it could be sybil'd easy enough by someone motivated but in general day to day use I think it might be beneficial.